File: bindings/php/src/ProvenWebdav.php
<?php
// SPDX-License-Identifier: PMPL-1.0-or-later
// Copyright (c) 2026 Jonathan D.A. Jewell (hyperpolymath) <j.d.a.jewell@open.ac.uk>
//
// WebDAV protocol types for proven-servers.

declare(strict_types=1);

namespace ProvenServers;

/** Method matching the Idris2 ABI tags. */
enum Method: int
{
    case Propfind = 0;
    case Proppatch = 1;
    case Mkcol = 2;
    case Copy = 3;
    case Move = 4;
    case Lock = 5;
    case Unlock = 6;
}

/** StatusCode matching the Idris2 ABI tags. */
enum StatusCode: int
{
    case MultiStatus = 0;
    case UnprocessableEntity = 1;
    case Locked = 2;
    case FailedDependency = 3;
    case InsufficientStorage = 4;
}

/** LockScope matching the Idris2 ABI tags. */
enum LockScope: int
{
    case Exclusive = 0;
    case Shared = 1;
}

/** LockType matching the Idris2 ABI tags. */
enum LockType: int
{
    case Write = 0;
}

/** Depth matching the Idris2 ABI tags. */
enum Depth: int
{
    case Zero = 0;
    case One = 1;
    case Infinity = 2;
}

/** PropertyOp matching the Idris2 ABI tags. */
enum PropertyOp: int
{
    case Set = 0;
    case Remove = 1;
}

File: bindings/csharp/src/ProvenWebdav.cs
// SPDX-License-Identifier: MPL-2.0
// Copyright (c) 2026 Jonathan D.A. Jewell (hyperpolymath) <j.d.a.jewell@open.ac.uk>
//
// WebDAV protocol types for proven-servers.

namespace Proven;

/// <summary>Method matching the Idris2 ABI tags (0-6).</summary>
public enum Method : byte
{
    Propfind = 0,
    Proppatch = 1,
    Mkcol = 2,
    Copy = 3,
    Move = 4,
    Lock = 5,
    Unlock = 6
}

/// <summary>StatusCode matching the Idris2 ABI tags (0-4).</summary>
public enum StatusCode : byte
{
    MultiStatus = 0,
    UnprocessableEntity = 1,
    Locked = 2,
    FailedDependency = 3,
    InsufficientStorage = 4
}

/// <summary>LockScope matching the Idris2 ABI tags (0-1).</summary>
public enum LockScope : byte
{
    Exclusive = 0,
    Shared = 1
}

/// <summary>LockType matching the Idris2 ABI tags (0-0).</summary>
public enum LockType : byte
{
    Write = 0
}

/// <summary>Depth matching the Idris2 ABI tags (0-2).</summary>
public enum Depth : byte
{
    Zero = 0,
    One = 1,
    Infinity = 2
}

/// <summary>PropertyOp matching the Idris2 ABI tags (0-1).</summary>
public enum PropertyOp : byte
{
    Set = 0,
    Remove = 1
}
